cmake/templates/setup.sh.in

```diff
@@ -7,6 +7,7 @@
 # --extend: skips the undoing of changes from a previously sourced setup file
 #   (in plain sh shell which does't support arguments for sourced scripts you
 #   can set the environment variable `CATKIN_SETUP_UTIL_ARGS=--extend` instead)
+: ${CATKIN_SETUP_UTIL_ARGS:=}
```

Comment:
If the env var `CATKIN_SETUP_UTIL_ARGS` is not defined it will be set after this script. Therefore I would recommend to use the variable below the same way you modified `TMPDIR` to avoid defining env var.